crates/core/src/operations/subscribe.rs

```diff
@@ -66,45 +65,48 @@ pub(crate) async fn request_subscribe(
     op_manager: &OpManager,
     sub_op: SubscribeOp,
 ) -> Result<(), OpError> {
-    let (target, _id) = if let Some(SubscribeState::PrepareRequest { id, key }) = &sub_op.state {
-        if !super::has_contract(op_manager, *key).await? {
-            tracing::debug!(%key, "Contract not found, trying other peer");
-            return Err(OpError::ContractError(ContractError::ContractNotFound(
-                *key,
-            )));
-        }
+    if let Some(SubscribeState::PrepareRequest { id, key }) = &sub_op.state {
+        // TODO: Handle local subscriptions properly
+        // Currently, if a contract is cached locally, subscriptions still go through
```

Comment:
This is incorrect, the network and local parts of the subscription are detached and happen independently and asynchronously, so there is no lose in efficiency. We already handle local subscriptions without network messages by design.

And with the current change, since we always put locally anyway, the clients can subscribe automatically to peer updates.

Reply:
This TODO and comment are incorrect and should be removed